network_simulator/plot_tool.py

```diff
@@ -0,0 +1,390 @@
+import matplotlib.pyplot as plt
+import numpy as np
+
+from common import *
+
+class PlotTool(object):
+    """
+    Produces graph of different data structures.
+
+    Windowing is used to group together data that occurred between a given
+    timestamp and the previous timestamp in the output list (where the
+    timestamps are regularly spaced apart in the output list; the constant
+    for spacing is defined in common.py).
+    """
+    @staticmethod
+    def gen_count_interval_list(lst):
+        """
+        Computes a tuple list of counts based on a timestamp list.
+
+        :param list lst: list of timestamps to be converted.
+        :return: list: a list of (time, count) tuples where the counts are
+        aggregated count of data within the window time.
+        """
+        # Count within one window interval.
+        curr_window_count = 0
+        # End point of each window interval.
+        curr_window_end_time = GRAPH_WINDOW_SIZE
+        output = []
+        for idx in range(len(lst)):
+            timestamp = lst[idx]
+            if timestamp > curr_window_end_time:
+                output.append((curr_window_end_time, curr_window_count))
+                curr_count = 0
```

Comment:
I think you meant to set curr_window_count = 0. Be careful about renaming in Python (use PyCharm's refactor where possible).